src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -1508,23 +1508,29 @@ void IslandWindow::_globalActivateWindow(const uint32_t dropdownDuration,
     }
     else
     {
-        const auto windowThreadProcessId = GetWindowThreadProcessId(oldForegroundWindow, nullptr);
-        const auto currentThreadId = GetCurrentThreadId();
-
-        LOG_IF_WIN32_BOOL_FALSE(AttachThreadInput(windowThreadProcessId, currentThreadId, true));
-        // Just in case, add the thread detach as a scope_exit, to make _sure_ we do it.
-        auto detachThread = wil::scope_exit([windowThreadProcessId, currentThreadId]() {
-            LOG_IF_WIN32_BOOL_FALSE(AttachThreadInput(windowThreadProcessId, currentThreadId, false));
-        });
-        LOG_IF_WIN32_BOOL_FALSE(BringWindowToTop(_window.get()));
-        ShowWindow(_window.get(), SW_SHOW);
-
-        // Activate the window too. This will force us to the virtual desktop this
-        // window is on, if it's on another virtual desktop.
-        LOG_LAST_ERROR_IF_NULL(SetActiveWindow(_window.get()));
-
-        // Throw us on the active monitor.
-        _moveToMonitor(oldForegroundWindow, toMonitor);
+        // Try first to send a message to the current foreground window. If it's not responding, it may
+        // be waiting on us to finsh launching. Give it only 50ms--a tiny time slice--so as to not wait
+        // too long before failing out. SendMessageTimeoutW returns nonzero if it succeeds.
+        if (0 != SendMessageTimeoutW(oldForegroundWindow, WM_NULL, 0, 0, SMTO_BLOCK | SMTO_ABORTIFHUNG, 50, nullptr))

```

Comment:
Is there any specific reason you chose 50? What if the system is currently under heavy load at 100% CPU usage and it takes longer than that to activate the other thread?

Reply:
I admit that this is a very small timeout. I didn't want to block Terminal's startup for a very long time. I don't know what the right choice is.

Reply:
Honestly because of `ABORTIFHUNG`, we could up it a little bit. We could consider using `SMTO_NOTIMEOUTIFNOTHUNG`, which indicates that _IF THE THREAD CAN RECEIVE MESSAGES,_ we will wait possibly-forever for it.

Reply:
I've changed it to 1000 now after all. I've used `SendMessageTimeoutToTopLevelWindowClasses` as a reference which uses 5000 despite using the same 3 flags.